x/crosschain/keeper/grpc_query_cctx_rate_limit.go

```diff
@@ -90,15 +93,46 @@ func (k Keeper) ListPendingCctxWithinRateLimit(c context.Context, req *types.Que
 		return uint32(len(cctxs)) >= limit
 	}
 
-	// query pending nonces for each foreign chain
+	// if a cctx falls within the rate limiter window
+	isCctxInWindow := func(cctx *types.CrossChainTx) bool {
+		// #nosec G701 checked positive
+		return cctx.InboundTxParams.InboundTxObservedExternalHeight >= uint64(leftWindowBoundary)
+	}
+
+	// query pending nonces for each foreign chain and get the lowest height of the pending cctxs
 	// Note: The pending nonces could change during the RPC call, so query them beforehand
```

Comment:
minor: is this commenttrue? My understanding is that the PRC call gets passes a context which gives a snapshot of the state during the RPC call. No other threads should be altering it. @lumtis is this correct?

Reply:
I wanted to know if snapshot is used here as well.

Reply:
Yes, the state doesn't change during the rpc call

Reply:
I removed the incorrect Note